lib/puppet/defaults.rb

```diff
@@ -58,6 +58,18 @@ def self.default_vendormoduledir
     end
   end
 
+  def self.default_cadir
+    return "" if Puppet::Util::Platform.windows?
+    old_ca_dir = "#{Puppet[:ssldir]}/ca"
+    new_ca_dir = '/etc/puppetlabs/puppetserver/ca'
+
+    if File.exists?(new_ca_dir)
```

Comment:
In the 7.x version, we do a check for the old dir being a symlink and then do a `readlink` here instead. Do we want to do that here too, or is this simpler logic more what we want? I didn't want to start returning a different location for anyone who previously might have had a symlink here... This seemed the most likely to change behavior for the fewest number of people, but maybe there's another edge case I haven't thought of that would make the `readlink` better.

Reply:
I feel like it would be valuable to test for the artifacts in the directory or the symlink from the old location to ensure that we've actually been fully migrated and someone didn't just make the directory for a random reason.

Reply:
Updated to read the symlink instead.